Firmware/Source/v1.2/grove_pi_v1_2_6/grove_pi_v1_2_6.ino

```diff
@@ -96,7 +96,7 @@ int j;
 void loop()
 {
   long dur,RangeCm;
-  if(index==4)
+  if(index==4 && flag==0)
```

Comment:
I am not 100% sure, but it seems that at some point in the firmware history, this flag test was removed and replaced with a "run_once" flag. 

This "flag==0" test ensured that all commands would be executed only once.
The "run_once" flag is used to protect only specific commands from continuous repetition.

When patching some time ago the firmware, I was under the impression that this change had been made so as to "spam" commands to the grove sensors (maybe to reduce the impact of i2c communication problems?), except for a select few that would not support such a behavior.

@karan259, could you clarify this point?